random1on1/random1on1bot.py

```diff
@@ -15,32 +23,139 @@ def read_config(location):
+        return self.guild.default_role
+
+    def get_random1on1_role(self):
+        random1on1_roles = [
+            r for r in self.guild.roles
+            if r.name == self.config.random1on1_role
+        ]
+        if len(random1on1_roles) == 0:
+            random1on1_role = self.guild.create_role(
+                name=self.config.random1on1_role,
+                mentionable=True,
+                reason=
+                f"Role: {self.config.random1on1_role} required for Random 1-on-1s did not exist, so I created it!",
+            )
+        elif len(random1on1_roles) == 1:
+            random1on1_role = random1on1_roles[0]
+        else:
+            raise RuntimeError(
+                f"Found multiple roles of name {self.config.random1on1_role}")
 
-    def generate_pairs(self):
-        # TODO: Generate the pairs according to proper algorithm
-        raise NotImplementedError()
+        return random1on1_role
 
-    def log_pairings(self, pairings, dry_run=False):
-        # TODO: Log pairings in proper channel
-        raise NotImplementedError()
+    def get_announcement_channel(self) -> AnnouncementChannel:
+        announcement_channel = AnnouncementChannel(
+            name=self.config.announcement_channel_name, category=self.category)
+        _ = announcement_channel.set_permissions(
+            default_role=self.default_role,
+            random1on1_role=self.random1on1_role)
+        return announcement_channel
 
-    def send_pairings(self, pairings, dry_run=False):
-        # TODO: Send out pairings
-        raise NotImplementedError()
+    def get_history_channel(self) -> HistoryChannel:
+        history_channel = HistoryChannel(name=self.config.history_channel_name,
+                                         category=self.category)
+        _ = history_channel.set_permissions(
+            default_role=self.default_role,
+            random1on1_role=self.random1on1_role)
+        return history_channel
+
+    def get_logging_channel(self) -> LoggingChannel:
+        logging_channel = LoggingChannel(name=self.config.logging_channel_name,
+                                         category=self.category)
+        _ = logging_channel.set_permissions(
+            default_role=self.default_role,
+            random1on1_role=self.random1on1_role)
+        return logging_channel
+
+    def get_participants(self) -> list[Participant]:
+        participants: list[Participant] = []
+        for member in self.get_random1on1_role().members:
+            participants.append(Participant(member))
+
+        return participants
 
     def run_matching_program(self, dry_run=False):
-        # TODO: Run matching program in full
-        raise NotImplementedError()
+
+        participants = self.get_participants()
+        previous_pairings = self.history_channel.read_historical_pairings()
+
+        # TODO: Make algorithms modular
```

Comment:
Make them as an input then.
Running it would look like:
```
algorithm = Some Algorithm(...)
bot = Random1on1Bot(..., algorithm)
bot.do_stuff()
...
```